### ISSUE TITLE
Config property to override base uri that is used by LRAClient to register to LRA

### ISSUE THREAD

**Issue by ochaloup** [eclipse/microprofile-lra#85]
When user declares method to be `@LRA` then there is started a LRA in background before the logic of the method is invoked. The LRA client exposes `complete` and `compensate` (and others) endpoints while the base url is taken from its hostname.

There are cases when it's needed to override the way how client enlists itself. It's for example case when the client and coordinator runs in docker and there is not appropriate that client exposes the compensate endpoint as `http://localhost:8080/compensate` (the `localhost` is where the client is bound at in docker image). But for coordinator can contact the client back the client needs to provide different name which can be routed by `docker network` back to the client. 
The same situation could be when client and coordinator are separated by a NAT or firewall etc.

**Comment by mmusgrov** [comment]
The base URL is taken from the JAX-RS request context of the @LRA method that was invoked, ie

```java
ContainerRequestContext#getUriInfo()#getBaseUri();
'''

**Comment by mmusgrov** [comment]
At the moment it is a URI:
```java
URL joinLRA(URL lraId, Class<?> resourceClass, URI baseUri, String compensatorData)
```
Are you saying that using a URI is insufficient? how will your proposal affect this particular API?

**Comment by ochaloup** [comment]
* The current behaviour is insufficient if you use annotations.
* The proposal does not affect this particular API in any way.
* Using URI in the `joinLRA` is sufficiant from my point of view.

When you define your method with `@LRA` and you define `@Complete`/`@Compensate` methods in the class then the LRA library will start the LRA instance and register this as participant. This participant will be registered as reachable at `containerRequestContext#getUriInfo()#getBaseUri()` (which is specific for Narayana LRA client and not defined in spec).

My proposal is to explicitly define this in spec and provide a way to override the base uri on the registration of participants when annotations are used.

**Comment by xstefank** [comment]
+1, this makes annotations and client interface functionally not same. We shouldn't be requiring code changes in user code when running in the docker.

**Comment by mmusgrov** [comment]
+1

**Comment by rdebusscher** [comment]
Is config property needed for 1.0? Since we removed interoperability and Java API for 1.0, locating the LRA coordinator is implementation specific.

**Comment by mmusgrov** [comment]
Good point, would there be any objections if I move this issue to 1.x

**Comment by ochaloup** [comment]
agree. this should be moved to 1.x until the interoperability will be addressed. for 1.0 release is this an implementation specific.
